Traceback (most recent call last):
  File "/opt/anaconda3/envs/codeforecon/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/opt/anaconda3/envs/codeforecon/lib/python3.8/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/anaconda3/envs/codeforecon/lib/python3.8/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/opt/anaconda3/envs/codeforecon/lib/python3.8/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/opt/anaconda3/envs/codeforecon/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/opt/anaconda3/envs/codeforecon/lib/python3.8/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/opt/anaconda3/envs/codeforecon/lib/python3.8/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/anaconda3/envs/codeforecon/lib/python3.8/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
df = pd.read_csv(
    "https://github.com/aeturrell/coding-for-economists/raw/main/data/smith_won.txt",
    delimiter="\n",
    names=["text"],
)
df.head()
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
Input [0;32mIn [3][0m, in [0;36m<cell line: 1>[0;34m()[0m
[0;32m----> 1[0m df [38;5;241m=[39m [43mpd[49m[38;5;241;43m.[39;49m[43mread_csv[49m[43m([49m
[1;32m      2[0m [43m    [49m[38;5;124;43m"[39;49m[38;5;124;43mhttps://github.com/aeturrell/coding-for-economists/raw/main/data/smith_won.txt[39;49m[38;5;124;43m"[39;49m[43m,[49m
[1;32m      3[0m [43m    [49m[43mdelimiter[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;130;43;01m\n[39;49;00m[38;5;124;43m"[39;49m[43m,[49m
[1;32m      4[0m [43m    [49m[43mnames[49m[38;5;241;43m=[39;49m[43m[[49m[38;5;124;43m"[39;49m[38;5;124;43mtext[39;49m[38;5;124;43m"[39;49m[43m][49m[43m,[49m
[1;32m      5[0m [43m)[49m
[1;32m      6[0m df[38;5;241m.[39mhead()

File [0;32m/opt/anaconda3/envs/codeforecon/lib/python3.8/site-packages/pandas/util/_decorators.py:311[0m, in [0;36mdeprecate_nonkeyword_arguments.<locals>.decorate.<locals>.wrapper[0;34m(*args, **kwargs)[0m
[1;32m    305[0m [38;5;28;01mif[39;00m [38;5;28mlen[39m(args) [38;5;241m>[39m num_allow_args:
[1;32m    306[0m     warnings[38;5;241m.[39mwarn(
[1;32m    307[0m         msg[38;5;241m.[39mformat(arguments[38;5;241m=[39marguments),
[1;32m    308[0m         [38;5;167;01mFutureWarning[39;00m,
[1;32m    309[0m         stacklevel[38;5;241m=[39mstacklevel,
[1;32m    310[0m     )
[0;32m--> 311[0m [38;5;28;01mreturn[39;00m [43mfunc[49m[43m([49m[38;5;241;43m*[39;49m[43margs[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m/opt/anaconda3/envs/codeforecon/lib/python3.8/site-packages/pandas/io/parsers/readers.py:663[0m, in [0;36mread_csv[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)[0m
[1;32m    660[0m [38;5;28;01mdel[39;00m kwds[[38;5;124m"[39m[38;5;124mfilepath_or_buffer[39m[38;5;124m"[39m]
[1;32m    661[0m [38;5;28;01mdel[39;00m kwds[[38;5;124m"[39m[38;5;124msep[39m[38;5;124m"[39m]
[0;32m--> 663[0m kwds_defaults [38;5;241m=[39m [43m_refine_defaults_read[49m[43m([49m
[1;32m    664[0m [43m    [49m[43mdialect[49m[43m,[49m
[1;32m    665[0m [43m    [49m[43mdelimiter[49m[43m,[49m
[1;32m    666[0m [43m    [49m[43mdelim_whitespace[49m[43m,[49m
[1;32m    667[0m [43m    [49m[43mengine[49m[43m,[49m
[1;32m    668[0m [43m    [49m[43msep[49m[43m,[49m
[1;32m    669[0m [43m    [49m[43merror_bad_lines[49m[43m,[49m
[1;32m    670[0m [43m    [49m[43mwarn_bad_lines[49m[43m,[49m
[1;32m    671[0m [43m    [49m[43mon_bad_lines[49m[43m,[49m
[1;32m    672[0m [43m    [49m[43mnames[49m[43m,[49m
[1;32m    673[0m [43m    [49m[43mprefix[49m[43m,[49m
[1;32m    674[0m [43m    [49m[43mdefaults[49m[38;5;241;43m=[39;49m[43m{[49m[38;5;124;43m"[39;49m[38;5;124;43mdelimiter[39;49m[38;5;124;43m"[39;49m[43m:[49m[43m [49m[38;5;124;43m"[39;49m[38;5;124;43m,[39;49m[38;5;124;43m"[39;49m[43m}[49m[43m,[49m
[1;32m    675[0m [43m[49m[43m)[49m
[1;32m    676[0m kwds[38;5;241m.[39mupdate(kwds_defaults)
[1;32m    678[0m [38;5;28;01mreturn[39;00m _read(filepath_or_buffer, kwds)

File [0;32m/opt/anaconda3/envs/codeforecon/lib/python3.8/site-packages/pandas/io/parsers/readers.py:1536[0m, in [0;36m_refine_defaults_read[0;34m(dialect, delimiter, delim_whitespace, engine, sep, error_bad_lines, warn_bad_lines, on_bad_lines, names, prefix, defaults)[0m
[1;32m   1530[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[1;32m   1531[0m         [38;5;124m"[39m[38;5;124mSpecified a delimiter with both sep and [39m[38;5;124m"[39m
[1;32m   1532[0m         [38;5;124m"[39m[38;5;124mdelim_whitespace=True; you can only specify one.[39m[38;5;124m"[39m
[1;32m   1533[0m     )
[1;32m   1535[0m [38;5;28;01mif[39;00m delimiter [38;5;241m==[39m [38;5;124m"[39m[38;5;130;01m\n[39;00m[38;5;124m"[39m:
[0;32m-> 1536[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[1;32m   1537[0m         [38;5;124mr[39m[38;5;124m"[39m[38;5;124mSpecified [39m[38;5;124m\[39m[38;5;124mn as separator or delimiter. This forces the python engine [39m[38;5;124m"[39m
[1;32m   1538[0m         [38;5;124m"[39m[38;5;124mwhich does not accept a line terminator. Hence it is not allowed to use [39m[38;5;124m"[39m
[1;32m   1539[0m         [38;5;124m"[39m[38;5;124mthe line terminator as separator.[39m[38;5;124m"[39m,
[1;32m   1540[0m     )
[1;32m   1542[0m [38;5;28;01mif[39;00m delimiter [38;5;129;01mis[39;00m lib[38;5;241m.[39mno_default:
[1;32m   1543[0m     [38;5;66;03m# assign default separator value[39;00m
[1;32m   1544[0m     kwds[[38;5;124m"[39m[38;5;124mdelimiter[39m[38;5;124m"[39m] [38;5;241m=[39m delim_default

[0;31mValueError[0m: Specified \n as separator or delimiter. This forces the python engine which does not accept a line terminator. Hence it is not allowed to use the line terminator as separator.
ValueError: Specified \n as separator or delimiter. This forces the python engine which does not accept a line terminator. Hence it is not allowed to use the line terminator as separator.

